Fix pcps threshold

odrisci opened the pull request:
[DO NOT MERGE YET!]

This is a relatively big pull request implementing the fixes discussed in #331 .

It is open for discussion as there are some changes in here that I'm not sure are exactly the way to go, but they do start to address the issue of code duplication in the pcps_acquisition adapters, and also apply the fix for the pcps acquisition threshold.

carlesfernandez commented:
I understand the key issue here is at  75c57e90e5fbec1e5ede6db7dfd11d4f962a700e I like it. See #336 about how we do proceed about removing code duplication and moving to a new data type nomenclature. All those things mixed together are hard to process!

odrisci commented:
So I've reverted to the gnss-sdr style naming schemes here, and also added some unit tests for the item type helpers

I do have a question on the difference between "ibyte" and "cbyte". It seems to me that the memory layout is identical in each case. The only difference is that the item size is one byte for ibyte and 2 bytes for cbyte. I don't see the use case for ibyte - the only difference is when you expect your source stream to send you an uneven number of samples (one extra in-phase sample), but I would prefer to always ensure that I got a matching Q sample for every I. Similar for "ishort". Surely it would be better to just enforce use of "byte" or "short" for real types and "cbyte" or "cshort" for complex types. Or am I missing something?

carlesfernandez commented:
Hi @odrisci 

you are right that "ibyte" and "ishort" make little sense, it is just a way to indicate that they are real samples. Ok to rename them to "byte" and "short" (with alias to the old names).

I've been working in this branch at [odrisci-fix_pcps_threshold](/gnss-sdr/gnss-sdr/tree/odrisci-fix_pcps_threshold) and I think it's ready to be merged. It passes all the tests in our CI without breaking existing configurations.

However, there is a change in the configuration. Now the `use_CFAR_algorithm` parameter is not exposed anymore to the configuration file, it is just an internal flag. If the user sets it, the setting is ignored. The usage is as follows:

 * If a manual `threshold` is set by the user (and `pfa` = 0, or not set), the first_vs_second_peak_statistic metric is used.  This is a fixed threshold.

 * If the `pfa` parameter is specified in the configuration file to a non-zero value, then the max_to_input_power_statistic metric is used, and the threshold is set (and recomputed as needed) automatically.
